packages/react-magma-dom/src/components/ToggleButton/ToggleButton.test.js

```diff
@@ -0,0 +1,197 @@
+        'border-radius',
+        magma.spaceScale.spacing03
+      );
+      expect(button).toHaveStyleRule('height', magma.spaceScale.spacing09);
+      expect(button).toHaveStyleRule('width', magma.spaceScale.spacing09);
+    });
+
+    it('Text only styling', () => {
+      const { getByTestId } = render(
+        <ToggleButton testId={testId}>{TEXT}</ToggleButton>
+      );
+      const button = getByTestId(testId);
+
+      expect(getByTestId(testId).querySelector('svg')).not.toBeInTheDocument();
+      expect(button).toHaveStyleRule('min-width', magma.spaceScale.spacing13);
+      expect(button).toHaveStyleRule('width', 'auto');
+    });
+
+    it('Text and icon styling', () => {
+      const { getByTestId, getByText } = render(
+        <ToggleButton icon={icon} testId={testId}>
+          {TEXT}
+        </ToggleButton>
+      );
+      const button = getByTestId(testId);
+
+      expect(getByTestId(testId).querySelector('svg')).toBeInTheDocument();
+      expect(getByText(TEXT)).toHaveStyleRule(
+        'padding-left',
+        magma.spaceScale.spacing03
+      );
+      expect(button).toHaveStyleRule('min-width', 'auto');
+    });
+
+    it('Consistent border styling on disabled', () => {
+      const { getByTestId, rerender } = render(
+        <ToggleButton testId={testId}>{TEXT}</ToggleButton>
+      );
+      const button = getByTestId(testId);
+
+      expect(button).toHaveStyleRule('border', '1px solid');
+      expect(button).toHaveStyleRule('border-color', magma.colors.neutral300);
+
+      rerender(
+        <ToggleButton testId={testId} disabled>
+          {TEXT}
+        </ToggleButton>
+      );
+
+      expect(button).toHaveStyleRule('border', '1px solid');
+      expect(button).toHaveStyleRule('border-color', magma.colors.neutral300);
+    });
+
+    it('Supports small icon sizes', () => {
+      const { getByTestId } = render(
+        <ToggleButton size={ButtonSize.small} icon={icon} testId={testId} />
+      );
+      const button = getByTestId(testId);
+
+      expect(button).toHaveStyleRule('width', magma.spaceScale.spacing07);
+      expect(button).toHaveStyleRule('height', magma.spaceScale.spacing07);
+    });
+
+    it('Supports medium icon sizes', () => {
+      const { getByTestId } = render(
+        <ToggleButton size={ButtonSize.medium} icon={icon} testId={testId} />
+      );
+      const button = getByTestId(testId);
+
+      expect(button).toHaveStyleRule('width', magma.spaceScale.spacing09);
+      expect(button).toHaveStyleRule('height', magma.spaceScale.spacing09);
+    });
+
+    it('Supports large icon sizes', () => {
+      const { getByTestId } = render(
+        <ToggleButton size={ButtonSize.large} icon={icon} testId={testId} />
+      );
+      const button = getByTestId(testId);
+
+      expect(button).toHaveStyleRule('width', magma.spaceScale.spacing11);
+      expect(button).toHaveStyleRule('height', magma.spaceScale.spacing11);
+    });
+  });
+
+  describe('Grouping', () => {
```

Comment:
We should have a separate file for ToggleButtonGroup tests as they are two separate components.